src/v/kafka/protocol/schemata/generator.py

```diff
@@ -309,6 +315,63 @@
     fetch_record_set=("batch_reader", None, "read_nullable_batch_reader()"),
 )
 
+# Declare some fields as sensitive. Utmost care should be taken to ensure the
+# contents of these fields are never made available over unsecure channels
+# (sent over an unencrypted connection, printed in logs, etc.).
+sensitive_map = {
+    "AlterConfigsRequestData": {
+        "Resources": {
+            # NOTE: this doesn't take into account whether the altered config
+            # is sensitive. Since there is no IsSensitive field in the
+            # AlterableConfig, assume all configs are sensitive.
+            "Configs": {
+                "Value": True,
+            },
+        },
+    },
+    "DescribeConfigsResponseData": {
+        "Results": {
+            # TODO: this doesn't take into account whether the altered config
+            # is sensitive. There is an IsSensitive field in the
+            # DescribeConfigsResourceResult, that we may be able to leverage.
+            # For now, treat all configs as sensitive.
+            "Configs": {
+                "Value": True,
+            },
+        },
+    },
+    "FetchResponseData": {
+        "Topics": {
+            "Partitions": {
+                "Records": True,
+            },
+        },
+    },
+    "IncrementalAlterConfigsRequestData": {
+        "Resources": {
+            # NOTE: this doesn't take into account whether the altered config
+            # is sensitive. Since there is no IsSensitive field in the
+            # AlterableConfig, assume all configs are sensitive.
+            "Configs": {
+                "Value": True,
```

Comment:
let's find out what sort of things we want to redact here and be more precise about what is left out. having details in the log about requests is valuable for debugging.

i'm not totally sure we let any sensitive data in through this and the other config related requests, but if we do, we could probably use a deny list here using the same basic approach with this map.

Reply:
Opened https://github.com/redpanda-data/redpanda/issues/4983 to track this

Reply:
Walking through the handler code for the config-related messages, it does indeed look like we only handle a subset of configs, and that subset is as a whole not sensitive. I've reverted the config redaction.